cli/install.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python
+
+import logging
+import os
+
+import yaml
+
+from cli import logger  # logger creation is first thing to be done
+from cli import conf
+from cli import exceptions
+from cli import utils
+import cli.yamls
+import cli.execute
+
+ENTRY_POINT = "installer"
+
+LOG = logger.LOG
+CONF = conf.config
+
+NON_SETTINGS_OPTIONS = ['command0', 'verbose', 'extra-vars', 'output-file',
+                        'input-files', 'dry-run', 'cleanup', 'inventory']
+
+
+def get_settings_dir(args=None):
+    """Retrieve settings dir.
+
+    :param args: Namespace of argument parser
+    :returns: path to settings dir. if args, return path to parser settings
+        dir. If args.command0 exists, return path to subparser settings dir.
+    """
+    settings_dir = utils.validate_settings_dir(
+        CONF.get('defaults', 'settings'))
+    if args:
+        settings_dir = os.path.join(settings_dir,
+                                    ENTRY_POINT)
+    if hasattr(args, "command0"):
```

Comment:
Is the command0 component is required in the it the settings path? I'd suggest to handle situation when command0 is not defined (e.g. raise an exception).